### PR TITLE
Default configuration for envoy gw

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/giantswarm/issues/32297
This PR:

- adds/changes/removes etc

### Testing

Description on how {APP-NAME} can be tested.

- [ ] fresh install works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM
- [ ] upgrade from previous version works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM

#### Other testing

Description of features to additionally test for {APP-NAME} installations.

- [ ] check reconciliation of existing resources after upgrading
- [ ] X still works after upgrade
- [ ] Y is installed correctly

### Checklist

- [ ] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/envoy-gateway-default-configuration/templates/envoy-proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                      readOnlyRootFilesystem: true
+                  securityContext:
+                    runAsGroup: 65534
+                    runAsNonRoot: true
+                    runAsUser: 65534
+                    seccompProfile:
+                      type: RuntimeDefault
+      envoyService:
+        annotations:
+          external-dns.alpha.kubernetes.io/hostname: ingress.mati208.gaws.gigantic.io
```
Comment (puja108) [inline helm/envoy-gateway-default-configuration/templates/envoy-proxy.yaml:40]: should get this from values right? sth like gateway.{basedomain}

Comment (puja108) [reply helm/envoy-gateway-default-configuration/templates/envoy-proxy.yaml:40]: below we use {{ .Values.gateway.host }}

File: helm/envoy-gateway-default-configuration/values.yaml [hunk 2]
```diff
@@ -7,6 +8,9 @@ project:
 
 image:
   registry: quay.io
-  name: giantswarm/{APP-NAME}
+  name: giantswarm/envoy-gateway-default-configuration
   tag: v0.0.1
   pullPolicy: IfNotPresent
+
+gateway:
+  host: test.gaws.gigantic.io
```
Comment (puja108) [inline helm/envoy-gateway-default-configuration/values.yaml:16]: not sure we should have a default here, that won't work right?

Comment (pipo02mix) [reply helm/envoy-gateway-default-configuration/values.yaml:16]: we have a dummy value like in IC

**Review by puja108 — APPROVED** [review]

**Pull request merged**